File: app/src/main/java/com/zxn/fuzzysearchdemo/ItemEntity.java
package com.zxn.fuzzysearchdemo;

import com.zxn.fuzzysearch.IFuzzySearchItem;

import java.util.List;

public class ItemEntity implements IFuzzySearchItem {

	private String       mValue;
	private String       mSortLetters;
	private List<String> mFuzzySearchKey;

	public ItemEntity(String value, String sortLetters, List<String> fuzzySearchKey) {
		mValue = value;
		mSortLetters = sortLetters;
		mFuzzySearchKey = fuzzySearchKey;
	}

	public String getValue() {
		return mValue;
	}

	@Override
	public String getSourceKey() {
		return mValue;
	}

	@Override
	public List<String> getFuzzyKey() {
		return mFuzzySearchKey;
	}
}

File: app/src/main/java/com/zxn/fuzzysearchdemo/ItemEntity.kt
package com.zxn.fuzzysearchdemo

import com.zxn.fuzzysearch.IFuzzySearchItem
/**
 *  Updated by zxn on 2020/11/2.
 */
class ItemEntity(
    val value: String,
    private val mSortLetters: String,
    private val mFuzzySearchKey: List<String>
) : IFuzzySearchItem {
    override val sourceKey: String?
        get() = value
    override val fuzzyKey: List<String?>?
        get() = mFuzzySearchKey

//    fun getSourceKey(): String {
//        return value
//    }
//
//    fun getFuzzyKey(): List<String> {
//        return mFuzzySearchKey
//    }

}